test/unit/component.js

```diff
@@ -82,142 +85,344 @@ describe('Component class', () => {
-        assert.deepEqual(component.styles, {product: {button: {color: 'red'}}});
       });
-    });
 
-    describe('get classes()', () => {
-      it('returns classes for each component in manifest', () => {
-        const component = new Component({id: 1234});
-        component.typeKey = 'product';
-        assert.equal(component.classes.product.product, 'shopify-buy__product');
-        assert.equal(component.classes.option.option, 'shopify-buy__option-select');
+      describe('unsuccessful initialization', () => {
+        let errorSetupModelStub;
+        let logNotFoundStub;
+
+        beforeEach(() => {
+          logNotFoundStub = sinon.stub(logNotFound, 'default');
+        });
+
+        afterEach(() => {
+          errorSetupModelStub.restore();
+          logNotFoundStub.restore();
+        });
+
+        it('catches and throws any error from setupModel', async () => {
+          const setupError = {message: ['test']};
+          errorSetupModelStub = sinon.stub(component, 'setupModel').rejects(setupError);
+          try {
+            await component.init(data);
+          } catch (error) {
+            assert.equal(error, setupError);
+          }
+          assert.throws(component.init, Error);
+        });
+
+        it('logs a not found error if the error message is "Not Found"', async () => {
+          const setupError = {message: ['Not Found']};
+          errorSetupModelStub = sinon.stub(component, 'setupModel').rejects(setupError);
+          try {
+            await component.init(data);
+          } catch (error) {
+            assert.equal(error, setupError);
+          }
+          assert.calledOnce(logNotFoundStub);
+          assert.calledWith(logNotFoundStub, component);
+        });
+
+        it('does not log a not found error if the error message is not "Not Found"', async () => {
+          const setupError = {message: ['Another Error']};
+          errorSetupModelStub = sinon.stub(component, 'setupModel').rejects(setupError);
+          try {
+            await component.init(data);
+          } catch (error) {
+            assert.equal(error, setupError);
+          }
+          assert.notCalled(logNotFoundStub);
+        });
       });
     });
 
-    describe('get selectors()', () => {
-      it('returns classes formatted as css selectors for each component in manifest', () => {
-        const component = new Component({id: 1234});
-        component.typeKey = 'product';
-        assert.equal(component.selectors.product.product, '.shopify-buy__product');
-        assert.equal(component.selectors.option.option, '.shopify-buy__option-select');
+    describe('setupModel()', () => {
+      it('returns passed data', async () => {
+        const model = await component.setupModel({test: 'test'});
+        assert.deepEqual(model, {test: 'test'});
+      });
+
+      it('fetches data if data was not passed', async () => {
+        component.fetchData = sinon.stub().resolves({test: 'fetchData'});
```

Comment:
Could we use this method of defining stubs (`const fetchDataStub = sinon.stub(component, 'fetchData').resolves(...);`) to be consistent with other tests?

Reply:
I tried that at first, but fetchData does not exist within component.js itself but its children. I was debating if it was worth to create a specific component that holds the fetchData function and stubbing it or approaching it like how it currently is.